tests/pgautofailover_utils.py

```diff
@@ -359,14 +359,21 @@ def run_sql_query(self, query, *args):
         Runs the given sql query with the given arguments in this postgres node
         and returns the results. Returns None if there are no results to fetch.
         """
-        with psycopg2.connect(self.connection_string()) as conn:
-            cur = conn.cursor()
-            cur.execute(query, args)
-            try:
-                result = cur.fetchall()
-                return result
-            except psycopg2.ProgrammingError:
-                return None
+        result = None
+        conn = psycopg2.connect(self.connection_string())
+
+        with conn:
```

Comment:
Do we still need to have a context handler on the connection object if the context handler doesn't close the connection anyway? What it is used for then?